test/libsolidity/syntaxTests/freeFunctions/free_call_via_contract_type.sol

```diff
@@ -0,0 +1,8 @@
+contract C {
+    function f() public pure {}
+}
+function fun() {
+    C.f();
+}
+// ----
+// TypeError 3419: (68-73): Cannot call function via contract type name.
```

Comment:
If it's pure, why not?

Reply:
You still need an instance. But you are right, most pure functions can probably be turned into free functions which then can be called without an instance.